shell/components/ResourceDetail/index.vue

```diff
@@ -1,569 +1,70 @@
-      if ( this.mode === _VIEW && this.as === _YAML && a && b && a !== b) {
-        this.yaml = await getYaml(this.$store, this.liveModel);
-      }
-    }
-  },
-
-  created() {
-    this.configureResource();
-  },
-
-  methods: {
-    stringify,
-    setSubtype(subtype) {
-      this.resourceSubtype = subtype;
-    },
-
-    keyAction(act) {
-      const m = this.liveModel;
-
-      if ( m?.[act] ) {
-        m[act]();
-      }
-    },
-    closeError(index) {
-      this.errors = this.errors.filter((_, i) => i !== index);
-    },
-    /**
-     * Initializes the resource components based on the provided user and
-     * resource override.
-     *
-     * Configures the detail and edit components for a resource based on the
-     * user's ID and the specified resource.
-     *
-     * @param {Object} user - The user object containing user-specific
-     * information.
-     * @param {string|null} resourceOverride - An optional resource override
-     * string. If not provided, the method will use the default resource from
-     * the route parameters or the instance's resourceOverride property.
-     */
-    configureResource(userId = '', resourceOverride = null) {
-      const id = userId || this.$route.params.id;
-      const resource = resourceOverride || this.resourceOverride || this.$route.params.resource;
-      const options = this.$store.getters[`type-map/optionsFor`](resource);
-
-      const detailResource = options.resourceDetail || options.resource || resource;
-      const editResource = options.resourceEdit || options.resource || resource;
-
-      // FIXME: These aren't right... signature is (rawType, subType).. not (rawType, resourceId)
-      // Remove id? How does subtype get in (cluster/node)
-      this.detailComponent = this.$store.getters['type-map/importDetail'](detailResource, id);
-      this.editComponent = this.$store.getters['type-map/importEdit'](editResource, id);
-    },
-    /**
-     * Sets the mode and initializes the resource components.
-     *
-     * This method sets the mode of the component and configures the resource
-     * components based on the provided user and resource.
-     *
-     * @param {Object} payload - An object containing the mode, user, and
-     * resource properties.
-     * @param {string} payload.mode - The mode to set.
-     * @param {Object} payload.user - The user object containing user-specific
-     * information.
-     * @param {string} payload.resource - The resource string to use for
-     * initialization.
-     */
-    setMode({ mode, userId, resource }) {
-      this.mode = mode;
-      this.value.id = userId;
-      this.configureResource(userId, resource);
-    }
-  }
+// Ideally I'd prefer to have separate routes/pages for each of these but our app makes a
+// fair amount of assumptions around having one detail page for each resource and that the
+// detail, config, edit, yaml, create pages are all derived from the same page.
```

Comment:
I think this is a fair tradeoff that preserves our existing routing and structure and better exposes which resources will be displayed and how. IMO this approach will be easier to debug in the future.